map/map_loader/src/pointcloud_map_loader/main.cpp

```diff
@@ -32,23 +32,23 @@
 
 #include <boost/filesystem.hpp>
 
-#include <ros/ros.h>
+#include <rclcpp/rclcpp.hpp>
 
 #include <map_loader/pointcloud_map_loader_node.h>
 
 namespace fs = boost::filesystem;
 
 int main(int argc, char * argv[])
 {
-  ros::init(argc, argv, "pointcloud_map_loader");
+  rclcpp::init(argc, argv);
 
   std::vector<std::string> pcd_paths;
   for (int i = 1; i < argc; ++i) {
     const fs::path arg(argv[i]);
 
     if (!fs::exists(arg)) {
       const std::string msg = "invalid path: " + arg.string();
-      throw std::runtime_error(msg);
+      std::cerr << msg;
```

Comment:
I changed this to std::cerr instead of throwing runtime_error exception.
This is because when I tried launching pointcloud_map_loader.launch.xml, it appends remap option as argument when it runs the actual command, making the node to terminate at this exception.

```
[INFO] [launch]: All log files can be found below /home/mitsudome-r/.ros/log/2020-10-20-21-45-27-613142-ade-9523
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [pointcloud_map_loader-1]: process started with pid [9525]
[pointcloud_map_loader-1] terminate called after throwing an instance of 'std::runtime_error'
[pointcloud_map_loader-1]   what():  invalid path: --ros-args
[ERROR] [pointcloud_map_loader-1]: process has died [pid 9525, exit code -6, cmd '/home/mitsudome-r/AutowareArchitectureProposal/install/map_loader/lib/map_loader/pointcloud_map_loader /home/mitsudome-r/AutowareArchitectureProposal/kashiwanoha/pointcloud_map.pcd --ros-args -r __node:=pointcloud_map_loader -r output/pointcloud_map:=/map/pointcloud_map'].
```

Reply:
Do you mean, the error condition should be changed? like below.

```
  for (int i = 1; i < argc; ++i) {
    const fs::path arg(argv[i]);

    if (fs::exists(arg)) break;
    if (i == argc - 1) {
        throw std::runtime_error("no valid path exist in args");
    }
}
```

Reply:
Yes, that's actually what I chose to do in the end. Instead of your code, I did checked if there are any valid paths in pcd_paths in https://github.com/tier4/Pilot.Auto/blob/3deb7d2741648315d5f7e30f7fe9d578e459039e/map/map_loader/src/pointcloud_map_loader/main.cpp#L70.

I just wanted the reviewers to know that the behavior of error check changed from ROS1.

Reply:
Understand. Thanks!